<launch>

  <arg name="vis"		default="true" />
  <arg name="board_ip" 		default="10.224.60.128" />
  <arg name="has_left_cam" 	default="true" />
  <arg name="has_right_cam" 	default="true" />
  <arg name="has_color_cam" 	default="true" />
  <arg name="stereo_cam_topic"  default="vrm3dvision" />
  <arg name="point_cloud_topic" default="point_cloud" />
  <arg name="point_cloud_trimmed_topic" default="point_cloud_trimmed" />
  <arg name="remove_outliers"   default="true" />
  <arg name="certainty_type"	default="0" />
  <arg name="pc_color_option"   default="none" />

  <arg name="layers"		default="10" />

        <group ns="/vrm3dvision/">
                <node pkg="vrm3dvision" type="vrm_log_node" name="VrmLogNode" output="screen">
			<param name="vrm_board_ip" value="$(arg board_ip)" />
                </node>

                <node pkg="vrm3dvision" type="vrm_command_node" name="VrmCommandNode" output="screen">
			<param name="initial_mode" value="11"/>
			<param name="initial_expo_grey" type="string" value="2.0" /> 
			<param name="initial_pattern_levels" value="$(arg layers)" />
			<param name="vrm_board_ip" value="$(arg board_ip)" />
                </node>

                <node pkg="vrm3dvision" type="vrm_image_proc" name="VrmImageProcNode" output="screen">
			<param name="vrm_board_ip" value="$(arg board_ip)" />
			<param name="has_left_camera" value="$(arg has_left_cam)" />
			<param name="has_right_camera" value="$(arg has_right_cam)" />
			<param name="has_color_camera" value="$(arg has_color_cam)" />
			<param name="stereo_camera_topic" value="vrm_stereo_camera" />
			<param name="color_cam_topic" value="color_cam" />
			<param name="visualization" value="$(arg vis)" />
			<param name="no_of_layers_structured_light" value="$(arg layers)" />
			<param name="point_cloud_pub_topic" value="$(arg point_cloud_topic)" />			
			<param name="point_cloud_trimmed_pub_topic" value="$(arg point_cloud_trimmed_topic)" />
			<param name="publish_trimmed_point_cloud" value="true" />			
			<param name="z_trim_distance" value="0.40" />
			<param name="pc_color_option" value="$(arg pc_color_option)" />
 			<param name="remove_outliers" value="$(arg remove_outliers)" />
			<param name="certainty_type" value="$(arg certainty_type)" />
 			<param name="mean_k" value="50" />
 			<param name="std_dev_thresh" value="8.0" />
                </node>		
		<node pkg="vrm3dvision" type="pcl_pointcloud_viewer" name="PCLPCviewer" output="screen">
			<param name="point_cloud_subscriber_topic" value="$(arg point_cloud_topic)" />
		</node> 

		<node pkg="rqt_service_caller" type="rqt_service_caller" name="RqtServiceCaller" />
        </group>
</launch>
